var/spack/repos/builtin/packages/babelstream/package.py

```diff
@@ -414,3 +572,232 @@ def cmake_args(self):
+                sys.exit(self.spec.variants["foption"].value + "is not supported by this compiler")
+
+        # ===================================
+        #               CRAY
+        # ===================================
+        if spec.compiler.name == "cce":
+            flags = "-e F -O3"
+
+            config["DOCONCURRENT_FLAG"] = "-h thread_do_concurrent -DCRAY_THREAD_DOCONCURRENT"
+            config["ARRAY_FLAG"] = "-h autothread"
+            config[
+                "OPENMP_FLAG"
+            ] = pkg.compiler.openmp_flag  # if clang based it will be -fopenmp else -h omp
+            config["OPENACC_FLAG"] = "-h acc"  # for cpu only -h omp
+
+            # Error Messages
+            if self.spec.variants["foption"].value in ["CUDA", "CUDAKernels"]:
+                sys.exit(self.spec.variants["foption"].value + "is not supported by this compiler")
+
+        # ===================================
+        #               GCC
+        # ===================================
+        if spec.compiler.name == "gcc":
+            flags = "-std=f2018 -O3 "
+            flags += "-Wall -Wno-unused-dummy-argument -Wno-unused-variable "
+
+            # config['MARCH'] = "neoverse-v1,neoverse-n1,icelake-server,znver3,cortex-a78ae"
+
+            if platform.machine() == "aarch64":
+                # TODO: Add MCPU option here as in -mcpu=$(MCPU)
+                flags += "-mcpu=native"
+            else:
+                # TODO: Add MARCH option here as in -march=$(MARCH)
+                flags += "-march=native"
+
+            config["FCFLAGS"] = flags
+            config["DOCONCURRENT_FLAG"] = "-ftree-parallelize-loops=4"
+            config["OPENMP_FLAG"] = pkg.compiler.openmp_flag
+            config["OPENACC_FLAG"] = "-fopenacc"
+
+            # Error Messages
+            if "CUDA" in self.spec.variants["foption"].value:
+                sys.exit(self.spec.variants["foption"].value + "is not supported by this compiler")
+
+        # ===================================
+        #               NVHPC
+        # ===================================
+        if spec.compiler.name == "nvhpc":
+            flags = pkg.compiler.opt_flags[4]  # for -O3
+            # FCFLAGS	:= -O3 -Minform=inform -Minfo=all
+            flags += " -Minform=warn "
+            TARGET = "gpu"  # target = "multicore"
+            config["TARGET"] = TARGET
+            if "cuda_arch" in self.spec.variants:
+                cuda_arch_list = self.spec.variants["cuda_arch"].value
+                # the architecture value is only number so append sm_ to the name
+                cuda_arch = "cc" + cuda_arch_list[0]
+            # config['MARCH'] = "neoverse-v1,neoverse-n1,icelake-server,znver3,cortex-a78ae"
+            GPUFLAG = " -gpu=" + cuda_arch
+            # march=neoverse-v1,neoverse-n1,zen3
+            march = "none"
+            if march != "none":
+                if platform.machine() == "aarch64":
```

Comment:
We should be keying off `spec.target` here, not `platform.machine()`.

`spec.target.family` gives the information at the granularity of `aarch64/x86_64/ppc64le`.